### DISCUSSION TITLE
Schema change for a deploy request is not deployable

### DISCUSSION THREAD

**Discussion by achie27** [planetscale/discussion#218] [Q&A] [answered]
Hey folks, I have the following diff on a table for a deploy request -
```
 CREATE TABLE `tokenprice` (
  `last_updated` bigint,
+ `id` char(32) NOT NULL,
  `token_symbol` varchar(255) NOT NULL,
  `currency` varchar(255) NOT NULL,
  `price` float NOT NULL,
- PRIMARY KEY (`currency`)
+ PRIMARY KEY (`id`),
+ UNIQUE KEY `token_symbol` (`token_symbol`, `currency`),
+ KEY `idx_token_symbol_currency` (`token_symbol`, `currency`)
 ) ENGINE InnoDB,
  CHARSET utf8mb4,
  COLLATE utf8mb4_0900_ai_ci;
```

But I am getting ```Table `tokenprice` has no non-null unique key.```
I don't understand why this is happening because `id` here is supposed to be a fit for a "non-null unique key" - can anyone share what might be the issue here?

**Comment by fooforge** [comment]
Hey @achie27,

I was able to reproduce this and am also a bit stumped why this isn't working. I'm going to write this up internally and will be in touch again once I have news to share.

**Reply by achie27** [reply]
Thanks @fooforge!
However, I encountered [this doc](https://docs.planetscale.com/learn/change-single-unique-key) later and turns out migrations like this need to be three-stepped. I was able to do my schema change when I fit the diff in the process outlined in the doc.

Do you think that's the only way to do such migrations or is there a simpler way?

**Reply by orware** [reply]
Thank you for sharing the Docs page reference @achie27 and that seems like the current recommended approach for such changes. 

It may be something that there is eventually a simpler approach for in the future, but at the moment, due to the reasons described on that Docs page, it is required.

**Comment by achie27** [comment, marked as answer]
I was able to do my schema change when I fit the diff in the process outlined in [this doc](https://docs.planetscale.com/learn/change-single-unique-key).

**Reply by fooforge** [reply]
I'm glad you were able to resolve this on your own, and I'm thankful for the learning opportunity. Much appreciated! ;-)
